app/lib/features/tasks/providers/notifiers.dart

```diff
@@ -86,3 +90,41 @@ class TaskNotifier extends FamilyAsyncNotifier<Task, Task> {
     return task;
   }
 }
+
+//List of all task list
+//If space Id given then it will return list of space task list
+class AsyncTaskListsNotifier
+    extends FamilyAsyncNotifier<List<TaskList>, String?> {
+  late Stream<bool> _listener;
+  late StreamSubscription<bool> _poller;
+
+  @override
+  Future<List<TaskList>> build(String? arg) async {
+    final client = ref.watch(alwaysClientProvider);
+
+    //GET ALL TASKS LIST
+    _listener = client.subscribeStream('tasks');
+
+    _poller = _listener.listen((e) async {
+      state = await AsyncValue.guard(() => _getTasksList(client));
+    });
+    ref.onDispose(() => _poller.cancel());
+
+    return await _getTasksList(client);
+  }
+
+  Future<List<TaskList>> _getTasksList(Client client) async {
+    //GET ALL TASKS LIST
+    final tasksList = (await client.taskLists()).toList();
+    // this might throw internally
```

Comment:
so... this is really not ideal as for two reasons: 1. we are loading _all tasks lists_ for every spaceId and the global one, holding them all in memory despite only needing them to be existing once and then filtered for whatever space we need, and 2. upon an update of `tasks` being issues by the client, _all_ of these notifiers fire and refresh again, despite only a subset actually needing to be refreshed.

Reply:
I see, understood the potential problem of above approach. Did the require changes as per below comment suggestions.